### PR TITLE
Updated to conform to new omf format.

### PR DESCRIPTION
**Pull request by toph-allen**
The new version of Oh My Fish doesn't read .load files, instead running an "init" function in the main .fish file of the plugin. I updated jump so that this works now.

### CODE REVIEW

File: jump.fish [hunk 1]
```diff
@@ -1,3 +1,10 @@
+function init -a path --on-event init_jump
+    set -gx MARKPATH $HOME/.marks
+    command mkdir -p $MARKPATH
+    complete -c jump -f -a '(command ls ~/.marks)'
+    complete -c unmark -f -a '(command ls ~/.marks)'
```
Comment (derekstavis) [inline jump.fish]: completions should go inside `completions/jump.fish` file.

**Comment by toph-allen** [comment]
Moved the completions to the appropriate location.

I originally had the `init` function in its own `init.fish` file, but that didn't seem to work. Putting it above the `jump` function definition in the `jump.fish` file works, and is in line with other packages I can find. Is that the intended format?

**Comment by derekstavis** [comment]
Thanks for taking time tor organize the PR. About `init` function not working in files other than `<package_name>.fish`, yes, this is the current format. It's an design issue, and is filed under https://github.com/oh-my-fish/oh-my-fish/issues/8.

File: completions/jump.fish [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# SYNOPSIS
+#   Completions for jump package.
+
+complete -c jump -f -a '(command ls ~/.marks)'
+complete -c unmark -f -a '(command ls ~/.marks)'
```
Comment (bobthecow) [inline completions/jump.fish]: add a newline to the end of this file?

Comment (toph-allen) [inline completions/jump.fish]: Done.

File: jump.fish [hunk 3]
```diff
@@ -1,3 +1,8 @@
+function init -a path --on-event init_jump
+    set -gx MARKPATH $HOME/.marks
```
Comment (bobthecow) [inline jump.fish]: Maybe add a check for not doing this line if it's already set?

**Comment by mrThe** [comment]
Any updates?

**Comment by toph-allen** [comment]
Crap, I'm sorry. Work's really busy. I'll address this on Tuesday.

File: jump.fish (hunk 3 above)
Comment (bpinto) [inline jump.fish]: Also, use two spaces for indentation instead of 4.

**Comment by bpinto** [comment]
Ping @toph-allen :)

**Comment by alvinlau** [comment]
very cool, waiting for this

**Comment by toph-allen** [comment]
Sorry about the delay. Travel, grant application, etc.

The plugin now checks to see if `$MARKPATH` is set before setting it. If it isn't set, it sets it to `$HOME/.marks` and `mkdir`s that directory.

And uses two-space indents consistently.

**Comment by derekstavis** [comment]
Would you mind squashing into a single commit?

**Comment by bpinto** [comment]
Thanks for the hard work @toph-allen

**Pull request merged**